Classes/FairPlayDRM/FPSContentKeyManager.swift

```diff
@@ -0,0 +1,88 @@
+import AVFoundation
+
+fileprivate let skdUrlPattern = try! NSRegularExpression(pattern: "URI=\"skd://([\\w-]+)\"", options: [])
+
+
+class FPSUtils {
+    static func extractAssetId(at location: URL) -> String? {
+        // Master should have the following line:
+        // #EXT-X-SESSION-KEY:METHOD=SAMPLE-AES,URI="skd://entry-1_x14v3p06",KEYFORMAT="com.apple.streamingkeydelivery",KEYFORMATVERSIONS="1"
+        // The following code looks for the first line with "EXT-X-SESSION-KEY" tag.
+        let re = skdUrlPattern
+        guard let master = try? String(contentsOf: location) else { PKLog.error("Can't read master playlist", location); return nil }
+        let lines = master.components(separatedBy: .newlines)
+        var assetId: String? = nil
+        
+        for line in lines {
+            if line.trimmingCharacters(in: .whitespaces).hasPrefix("#EXT-X-SESSION-KEY") {
+                guard let match = re.firstMatch(in: line, options: [], range: NSMakeRange(0, line.count)) else { continue }
+                if match.numberOfRanges < 2 { continue }
+                let assetIdRange = match.range(at: 1)
+                let start = line.index(line.startIndex, offsetBy: assetIdRange.location)
+                let end = line.index(line.startIndex, offsetBy: assetIdRange.location + assetIdRange.length - 1)
+                assetId = String(line[start...end])
+                
+                return assetId
+            }
+        }
+        
+        return nil
+    }
+
+    static func removeOfflineLicense(for location: URL, dataStore: LocalDataStore) -> Bool {
+        guard let id = extractAssetId(at: location) else {return false}
+        
+        do {
+            try dataStore.removeFpsKey(id)
+            return true
+        } catch {
+            return false
+        }
+    }
+    
+    static func checkOfflineLicense(for location: URL, dataStore: LocalDataStore) -> Date? {
```

Comment:
The name of the function is not clear. 
I can't know from the name that it returns the expiration date, only from the code.